init.lua

```diff
@@ -78,7 +78,11 @@ end
 
 
 local function IsDeathLinkEnabled()
-	return slot_options.death_link and ModSettingGet("archipelago.death_link")
+	local slot_death = nil
+	if slot_options.death_link ~= 0 and slot_options.death_link ~= nil then
+		slot_death = true
+	end
+	return slot_death and ModSettingGet("archipelago.death_link")
```

Comment:
Just realizing what you were saying before, and I guess 0 is not falsy like in other languages, but this should be better

```suggestion
	return slot_options.death_link == 1 and ModSettingGet("archipelago.death_link")
```